pkg/features/features.go

```diff
@@ -0,0 +1,68 @@
+package features
+
+import "fmt"
+
+type Feature struct {
+	Name        string
+	Description string
+	Enabled     bool
+}
+
+type FeatureSet struct {
+	Features     map[string]Feature
+	everythingOn bool
+}
+
+func NewFeatureSet() *FeatureSet {
+	return &FeatureSet{
+		Features:     make(map[string]Feature),
+		everythingOn: false,
+	}
+}
+
+func (fs *FeatureSet) AddFeature(name string, description string, enabled bool) {
```

Comment:
Nit: if you ever need to use https://pkg.go.dev/io/fs in here one or the other will need to be renamed

Reply:
Great point!